include/nbl/video/IAPIConnection.h

```diff
@@ -0,0 +1,31 @@
+#ifndef __NBL_I_API_CONNECTION_H_INCLUDED__
+#define __NBL_I_API_CONNECTION_H_INCLUDED__
+
+#include "nbl/core/IReferenceCounted.h"
+#include "nbl/video/IPhysicalDevice.h"
+#include "nbl/video/EApiType.h"
+
+namespace nbl {
+namespace video
+{
+
+class IAPIConnection : public core::IReferenceCounted
+{
+public:
+    // TODO implement in some source file in src/nbl/...
+    static core::smart_refctd_ptr<IAPIConnection> create(E_API_TYPE apiType, uint32_t appVer, const char* appName);
```

Comment:
still no, I want the Base classes to implement it and this not to be here

Reply:
? i don't get it
what's wrong? and what are base classes? in this context

Reply:
`COpenGLConnection`,`COpenGLESConnection`,`CVulkanConnection` etc.

then the headers for these should be in `include/nbl/video` and user should create them directly without `IAPIConnection`

(makes engine easier to extend with 3rdparty APIs, NVN, GNM, D3D12 etc.)

Reply:
I guess @achalpandeyy can fix on the Vulkan branch

Reply:
I fixed it in #182